src/shims/posix/fs.rs

```diff
@@ -22,15 +22,42 @@ struct FileHandle {
     writable: bool,
 }
 
+trait FileDescriptor<'tcx> : std::fmt::Debug {
```

Comment:
Why does the trait depend on `'tcx`? That should not be necessary. It would be better to make each method take a `'tcx` lifetime parameter.